src/pvv/side/door.php

```diff
@@ -0,0 +1,74 @@
+<?php
+namespace pvv\side;
+
+use \PDO;
+
+class Door{
+	private $pdo;
+
+	public function __construct(PDO $pdo){
+		$this->pdo = $pdo;
+	}
+
+	public function getAll() {
+		$query = 'SELECT time, open FROM door ORDER BY time DESC';
+		$statement = $this->pdo->prepare($query);
+		$statement->execute();
+
+		$doorEvents = [];
+		foreach($statement->fetchAll() as $row){
+			$doorEvents[] = [
+				'time'        => $row['time'],
+				'open'        => $row['open']
+			];
+		}
+
+		return $doorEvents;
+	}
+
+	public function getEntriesAfter($startTime) {
+		$query = 'SELECT time, open FROM door WHERE time > :startTime ORDER BY time DESC';
+		$statement = $this->pdo->prepare($query);
+		$statement->bindParam(':startTime', $startTime, PDO::PARAM_STR);
+		$statement->execute();
+
+		$doorEvents = [];
+		foreach($statement->fetchAll() as $row){
+			$doorEvents[] = [
+				'time'        => (int)$row['time'],
+				'open'        => (bool)$row['open']
+			];
+		}
+
+		return $doorEvents;
+	}
+
+	public function getCurrent() {
```

Comment:
What happens here if the table is empty?

Reply:
Tested and checked, door.php exhbits the preferred behaviour even with an empty table; getCurrent() returns time=0 and open=false,  and getEntriesAfter returns entries=[].

Reply:
So i assume $row is an empty map, and accessing it returns null, which gets cast into 0 and False?

Reply:
Correct, I check all types coming in, and extensively cast everything to the correct type before JSON encoding. Only ints(here cast to 0) and bools(here cast to false) can be returned as far as I can see.